actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -50,8 +50,19 @@ def assert_response(type, message = nil)
       #
       #   # Asserts that the redirection matches the regular expression
       #   assert_redirected_to %r(\Ahttp://example.org)
-      def assert_redirected_to(options = {}, message = nil)
-        assert_response(:redirect, message)
+      #
+      #   # Asserts that the redirection has the HTTP status code 301 (Moved
+      #   # Permanently).
+      #   assert_redirected_to "/some/path", status: :moved_permanently
+      def assert_redirected_to(options = {}, message = nil, kwargs = {})
```

Comment:
Hum, this one is tricky. All minitest `assert_` methods take `message` as last argument AFAIK. 

Also `options` here isn't particularly well named, it should be `url_options` I think.

So the signature should probably be `def assert_redirected_to(url_options = {}, options = {}, message = nil)`.

Then on the very first line you can probably do something like:

```ruby
  if message.is_a?(Hash) && options.empty?
    options, message = message, nil
  end
```

Reply:
> All minitest assert_ methods take message as last argument AFAIK.

I'm happy to make the suggested change, but the reason I went with keyword arguments is that there are existing examples of this approach.

```
activesupport/lib/active_support/testing/assertions.rb
175        def assert_changes(expression, message = nil, from: UNTRACKED, to: UNTRACKED, &block)
222        def assert_no_changes(expression, message = nil, from: UNTRACKED, &block)

activesupport/lib/active_support/testing/method_call_assertions.rb
9            def assert_called(object, method_name, message = nil, times: 1, returns: nil, &block)
51           def assert_called_on_instance_of(klass, method_name, message = nil, times: 1, returns: nil)
```

Reply:
In that case I'm fine if you actually use keyword arguments. Here you use an option hash, which isn't the same thing and make the argument parsing complicated for now reason.

So ok for `def assert_redirected_to(url_options = {}, message = nil, status: nil)`